test/org/javarosa/core/model/utils/test/DateUtilsTests.java

```diff
@@ -331,6 +334,40 @@ private void testCycle(Date in) {
 		}
 
 	}
+
+	public void testFormatting() {
+	    class LangJanSun {
+            private LangJanSun(String language, String january, String sunday) {
+                this.language = language;
+                this.january = january;
+                this.sunday = sunday;
+            }
+
+            private String language;
+	        private String january;
+	        private String sunday;
+        }
+
+        LangJanSun langJanSuns[] = new LangJanSun[] {
```

Comment:
I don't think this is particularly important, but it seemed like a relevant comment.

Personally I'd favour something like:
```
testFormatTranslations() {
	testFormatTranslation("en", "Sun Jan");
	testFormatTranslation("es", "dom ene");
	testFormatTranslation("fr", "dim. janv.");
}

testFormatTranslation(String language, String expectedJanSun) {
	Locale savedLocale = Locale.getDefault();

	
	Locale.setDefault(Locale.forLanguageTag(language));

	assertEquals(expectedJanSun, DateUtils.format(SUN_JAN_1_2017, "%a %b"));


	Locale.setDefault(savedLocale);
}
```